worlds/jakanddaxter/regs/BoggySwampRegions.py

```diff
@@ -0,0 +1,171 @@
+from typing import List
+
+from BaseClasses import CollectionState
+from .RegionBase import JakAndDaxterRegion
+from .. import EnableOrbsanity, JakAndDaxterWorld
+from ..Rules import can_fight, can_reach_orbs_level
+
+
+def build_regions(level_name: str, world: JakAndDaxterWorld) -> List[JakAndDaxterRegion]:
+    multiworld = world.multiworld
+    options = world.options
+    player = world.player
+
+    # This level is full of short-medium gaps that cannot be crossed by single jump alone.
+    # These helper functions list out the moves that can cross all these gaps (painting with a broad brush but...)
+    def can_jump_farther(state: CollectionState, p: int) -> bool:
+        return state.has_any({"Double Jump", "Jump Kick"}, p) or state.has_all({"Punch", "Punch Uppercut"}, p)
+
+    def can_jump_higher(state: CollectionState, p: int) -> bool:
+        return (state.has("Double Jump", p)
+                or state.has_all({"Crouch", "Crouch Jump"}, p)
+                or state.has_all({"Crouch", "Crouch Uppercut"}, p)
+                or state.has_all({"Punch", "Punch Uppercut"}, p))
```

Comment:
```suggestion
    def can_jump_higher(state: CollectionState, p: int) -> bool:
        return (state.has("Double Jump", p)
                or state.has_all(("Crouch", "Crouch Jump"), p)
                or state.has_all(("Crouch", "Crouch Uppercut"), p)
                or state.has_all(("Punch", "Punch Uppercut"), p))
```
I've been told it's very slightly faster for these to be tuples instead of sets. Up to you if you want to make that change, though.

Also this could technically be changed to check Crouch only once, but it's probably easier to read as it currently is.